airflow/providers/docker/operators/docker_swarm.py

```diff
@@ -123,11 +129,43 @@ def _run_image(self):
 
         self.log.info('Service started: %s', str(self.service))
 
-        status = None
         # wait for the service to start the task
         while not self.cli.tasks(filters={'service': self.service['ID']}):
             continue
-        while True:
+
+        logs = self.cli.service_logs(
+            self.service['ID'], follow=True, stdout=True, stderr=True, is_tty=self.tty
+        )
+        line = ''
+        _stream_logs = self.enable_logging  # Status of the service_logs' generator
+        while True:  # pylint: disable=too-many-nested-blocks
+            if self.enable_logging:
```

Comment:
Why do we need so many nested blocks?  Can you also create a new method that will not be performed when self.enable_logging == False?  When self.enable_logging = False, it is not necessary to call `self.cli.service_logs` also.

Reply:
Okay, I can think of a way to simplify the logic.

(By putting the service termination check within the `except ConnectionError` clause itself.)

Reply:
@mik-laj Thank you for the suggestions for improvement.

I have fixed these now. Please see if they look good to you.

Reply:
Thank you very much for this change. The code looks much better now.